runtime/jcl/common/mgmtinit.c

```diff
@@ -886,23 +888,29 @@ initMemoryManagement(J9JavaVM *vm)
 			mgmt->nonHeapMemoryPools[idx].id = J9VM_MANAGEMENT_POOL_NONHEAP_SEG_CLASSES;
 			strcpy(mgmt->nonHeapMemoryPools[idx].name, J9VM_MANAGEMENT_NONHEAPPOOL_NAME_CLASSES);
 			segList = vm->classMemorySegments;
+			mgmt->nonHeapMemoryPools[idx].maxSize = -1;
 			break;
 		case MISC_MEMORY:
 			mgmt->nonHeapMemoryPools[idx].id = J9VM_MANAGEMENT_POOL_NONHEAP_SEG_MISC;
 			strcpy(mgmt->nonHeapMemoryPools[idx].name, J9VM_MANAGEMENT_NONHEAPPOOL_NAME_MISC);
 			segList = vm->memorySegments;
+			mgmt->nonHeapMemoryPools[idx].maxSize = -1;
 			break;
 		case JIT_CODECACHE:
 			mgmt->nonHeapMemoryPools[idx].id = J9VM_MANAGEMENT_POOL_NONHEAP_SEG_JIT_CODE;
 			strcpy(mgmt->nonHeapMemoryPools[idx].name, J9VM_MANAGEMENT_NONHEAPPOOL_NAME_JITCODE);
 			segList = vm->jitConfig->codeCacheList;
+			mgmt->nonHeapMemoryPools[idx].maxSize = vm->jitConfig->codeCacheTotalKB * 1024;
 			break;
 		case JIT_DATACACHE:
-		default:
 			mgmt->nonHeapMemoryPools[idx].id = J9VM_MANAGEMENT_POOL_NONHEAP_SEG_JIT_DATA;
 			strcpy(mgmt->nonHeapMemoryPools[idx].name, J9VM_MANAGEMENT_NONHEAPPOOL_NAME_JITDATA);
 			segList = vm->jitConfig->dataCacheList;
+			mgmt->nonHeapMemoryPools[idx].maxSize = vm->jitConfig->dataCacheTotalKB * 1024;
 			break;
+		default:
+			/* Unreachable */
+			assert(FALSE);
```

Comment:
This should use a a trace library assert like `Assert_JCL_unreachable` rather than a c stdlib assert.  The `include assert.h` can be removed as well then